src/TestCases/Traits/AssertTrait.php

```diff
@@ -0,0 +1,38 @@
+<?php
+declare(strict_types=1);
+
+namespace Eonx\TestUtils\TestCases\Traits;
+
+use Eonx\TestUtils\Constraints\ArraySameWithDates;
+use Eonx\TestUtils\TestCases\Exceptions\InvalidParentClassException;
+use Eonx\TestUtils\TestCases\TestCase;
+
+trait AssertTrait
+{
+    /**
+     * AssertTrait constructor.
+     */
+    public function __construct()
+    {
+        if ($this instanceof TestCase === false) {
+            throw new InvalidParentClassException('AssertTrait must be used by an Eonx\\TestUtils\\TestCases class.');
```

Comment:
Why?

Reply:
not sure, thus I raised the question (don't know why its not showing here). What if a class that is not extending `TestCase` uses this trait? In that case `parent::assertThat()` won't work.

Reply:
I reckon if we make it checks for `PHPUnit\Framework\Assert` (which is where `assertThat()` lives) then we're good. That way this trait has maximum re-usability, and if need to include it in one of our other test cases, it still works fine.

Reply:
You only need the trait user to implement assertThat, so just have an abstract `assertThat` function on our Assert trait